mux_test.go

```diff
@@ -288,6 +289,52 @@ func TestMux(t *testing.T) {
 
 	expectUpdate(persistence.InvoiceStateSettled)
 
+	cancelSubscription()
+
+	// Create another new invoice for overpayment tests
+	invoice, testPreimage, err = creator.Create(
+		20000, time.Minute, "test 3", nil, 40,
+	)
+	require.NoError(t, err)
+	require.NotNil(t, invoice)
+
+	testHash = testPreimage.Hash()
+
+	cancelSubscription, err = registry.Subscribe(testHash, func(update InvoiceUpdate) {
+		logger.Sugar().Infow("Payment received", "state", update.State)
+
+		updateChan <- update
+	})
+	require.NoError(t, err)
+
+	// Regenerate onion blob for new hash.
+	onionBlob = genOnion()
+
+	// Expect failure for MPP overpayment
+	// TODO(aakselrod): decrypt and check failure message
```

Comment:
You want to do this in this PR?

An alternative would be to test overpayment in the invoice registry test, because it doesn't require decoding.

Reply:
I tried the latter and it came out much cleaner!